src/Squirrel/UpdateManager.ApplyReleases.cs

```diff
@@ -35,7 +35,20 @@ public ApplyReleasesImpl(string rootAppDirectory)
 
                 progress(0);
                 status("Applying updates");
-                
+
+                // Try to remove older versions before applying releases
+                try
+                {
+                    var currentVersion = updateInfo.CurrentlyInstalledVersion != null ?
+                        updateInfo.CurrentlyInstalledVersion.Version : null;
+
+                    await removeOldVersionFolders(new SemanticVersion[] { currentVersion });
+                }
+                catch (Exception ex)
+                {
+                    this.Log().WarnException("Failed to remove old folders, continuing anyways", ex);
```

Comment:
```suggestion
                    this.Log().WarnException("Failed to remove old folders.", ex);
```

Reply:
Actually, it's just one version we're trying to remove, right?

Reply:
Yes, in theory there can be more versions. This comment was copied from the original code ;)